### PR TITLE
Preflight Check

### PR DESCRIPTION
**Pull request by marek-vysoky**
Adding comments to PreflightCheck, writing unit test PreflightCheck_ValidResponse_ValidStatus

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @marek-vysoky has signed the CLA. Thanks for the pull request!

File: Box.V2/Managers/BoxFilesManager.cs [hunk 1]
```diff
@@ -101,7 +111,10 @@ public BoxFilesManager(IBoxConfig config, IBoxService service, IBoxConverter con
             request.ContentType = Constants.RequestParameters.ContentTypeJson;
 
             IBoxResponse<BoxPreflightCheck> response = await ToResponseAsync<BoxPreflightCheck>(request).ConfigureAwait(false);
-
+            if (response.ResponseObject != null)
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: Under which condition the ResponseObject == null?

**Comment by marek-vysoky** [comment]
I've deleted unnecessary validation

**Comment by coolboy** [comment]
@marek-vysoky conflict

**Comment by marek-vysoky** [comment]
Conflict resolved

File: Box.V2/Models/BoxPreflightCheck.cs [hunk 2]
```diff
@@ -18,11 +18,13 @@ public class BoxPreflightCheck
         /// The upload URL to optionally use when uploading the file
         /// </summary>
         [JsonProperty(PropertyName = FieldUploadUrl)]
+        [Obsolete]
```
Comment (coolboy) [inline Box.V2/Models/BoxPreflightCheck.cs]: So UploadUri & UploadToken will be always null in the response?

Should we mark FieldUploadUrl & FieldUploadToken as [Obsolete] as well?

ref:
https://docs.box.com/reference#preflight-check

File: Box.V2/Models/BoxPreflightCheck.cs [hunk 3]
```diff
@@ -35,6 +37,12 @@ public Uri UploadUri
         /// Currently not used.
         /// </summary>
         [JsonProperty(PropertyName = FieldUploadToken)]
+        [Obsolete]
         public string UploadToken { get; private set; }
+
+        /// <summary>
+        /// True if the upload would be successful;
+        /// </summary>
+        public bool Success;
```
Comment (coolboy) [inline Box.V2/Models/BoxPreflightCheck.cs]: why there is no { get; set; }

**Comment by marek-vysoky** [comment]
@coolboy  BoxPreflightCheck.Success is property, FieldUploadUrl and FieldUploadToken is marked as obsolete

**Pull request merged**